egs/wsj/s5/steps/nnet3/chain/make_weighted_den_fst.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# Copyright 2014-17 Vimal Manohar
+#           2017    Pegah Ghahremani
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+# WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+# MERCHANTABLITY OR NON-INFRINGEMENT.
+# See the Apache 2 License for the specific language governing permissions and
+# limitations under the License.
+
+# This script creates denominator FST (den.fst) and normalization.fst for
+# chain training. It additional copies the transition model and tree from the
+# first alignment directory to the chain directory.
+# This script can accept multiple sources of alignments with same phone sets
+# that can be weighted to estimate phone LM.
+# Each alignment directory should contain tree, final,mdl and ali.*.gz.
+
+set -o pipefail
+
+# begin configuration section.
+cmd=run.pl
+stage=-10
+weights= # comma-separated list of integer valued scale weights used
+         # to scale different phone sequences for different alignments.
+lm_opts='num_extra_lm_state=2000'
+#end configuration section.
+
+help_message="Usage: "$(basename $0)" [options] <ali-dir1> [<ali-dir2> ...] <out-dir>
```

Comment:
You can move the variable help_message to below parse_options.sh. Then it might be fine.